vllm/model_executor/model_loader/loader.py

```diff
@@ -165,16 +166,32 @@ def _process_weights_after_loading(model: nn.Module, model_config: ModelConfig,
             # parameters onto device for processing and back off after.
             with device_loading_context(module, target_device):
                 quant_method.process_weights_after_loading(module)
+            # There are cases of excess memory allocation that are not
+            # cleared, leading to "cuda out of memory" errors.
+            # TODO: Investigate the cause of the memory leak. It may be
+            # related to specific Python versions.
+            gc.collect()
```

Comment:
ideally we can run this just for every layer. but we don't have the concept of `layer` here.